### PR TITLE
parity-tests: restore twenty fixtures #1605's fold dropped, and un-excuse the launcher sequence

### PR DESCRIPTION
**Pull request by youknowone**
#1605 folded 79 parity fixtures into the benches and snippets it said already owned their shape. It merged with **no automated review** — CodeRabbit skipped it (146 files, over its 100 limit) and the Codex parity reviewer exited 1 on a usage limit — so nothing checked whether the coverage actually arrived. An audit of the merged diff finds twenty fixtures whose assertions did not, in three kinds.

### Six were deleted naming no destination at all

Nothing in the merge gained their content.

| fixture | what it owned |
|---|---|
| `format_dispatch_shapes.py` | the `__objclass__` receiver-owner rows — `is_shared_builtin_format` (`type_methods.rs`) consults `__objclass__` before formatting through a shared builtin `__format__` body, and `class S(str): __format__ = int.__format__` is the row that proves it. Nothing in the tree reads that path now. |
| `kwdefaults_dict_object_surface.py` | 12 drivers over the mapping behind `__kwdefaults__`: non-str-key devolution, `\|=`, `popitem`, pickle/deepcopy, mutation-during-iteration |
| `divmod_int_jit.py` | every decline arm of the exact-int `divmod` fold, incl. `divmod(INT_MIN, -1)` — a hard SIGFPE detector, not a printed value |
| `divmod_long_int_jit.py` | `rbigint.int_divmod`'s three allocations under `run.py --gc-poison`; poison exists only in the parity runner, so `pyre/bench` cannot substitute |
| `long_int_div_mod_jit.py` | reflected `int // long`, negative divisors, in-place forms |
| `long_int_pow_jit.py` | zero/negative/bool exponent, 3-arg `pow` in a hot loop |

### Thirteen reached a destination that took only part of them

Arms with no owner left include `call_defaults_inline.py`'s three-defaulted signature — the only one that separates `defs_w[p - def_first]` from a tail walk by more than one slot, and so the only reader of `inline_call.rs`'s array-backed `DefaultsRepr::ItemsBlock` arm at a nonzero index; four of `getattr_hook_inline_deopt.py`'s six compiled-hook deopt arms, including the hook that stores onto its own receiver; and `raise_bare_class_slot_defaults.py`'s per-iteration `args` / `__cause__` / `__context__` / `__suppress_context__` pins.

They come back as fixtures rather than as additions to the benches that absorbed the rest: a bench carries three recorded `.jitstats`, and these arms are correctness assertions, not measurements. What the benches took in the fold stays there.

### The twentieth is a live gate defect

The fold moved `launcher_prompt_entry_failure_reports.py`'s two arms into `launcher_prompt_entry_sequence.py` and carried its `# pyre-check: pypy-diverges:` header up to line 1 — which excuses the **whole** file, the ~110 lines of launcher contract that had been cross-checked against pypy until then. The two arms go back into their own file and the sequence fixture loses the header; it reports `pypy=OK`, so it had nothing to be excused for.

### Verification

`pyre/extra_tests/parity_tests/run.py --dynasm-only`: all twenty-one report `cpython=OK dynasm=OK`, and the five that declare a divergence report `pypy=xfail` — none `XPASS`. The single failure in the run is `foriter_stopiteration_exception_event.py [pypy]`, which fails under local pypy 7.3.20 and passes under the 3.11.15 CI runs; it is untouched here.

`check.py --backend dynasm --synthetic-only` 516/516 on this base (no bench or `.jitstats` is touched by this change).

The audit also cleared 7 further candidate findings as refuted, and 7 could not be verified before the reviewing run hit its own limit.

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved correctness for `sys.exc_info()` results, including cases where traceback information is not needed.
  * Improved handling of big-integer arithmetic, unary operations, exception propagation, attribute hooks, iteration, and function defaults.
  * Preserved expected behavior for type names, native formatting, math errors, filesystem operations, and interactive startup failures.

* **Tests**
  * Added broad parity coverage for edge cases involving exceptions, division, power operations, built-in types, and garbage collection.
  * Expanded benchmarks and validation across supported runtime configurations.

### CODE REVIEW

File: pyre/pyre-jit-trace/src/jitcode_dispatch/specialize.rs [hunk 1]
```diff
@@ -8617,6 +8625,189 @@ pub(crate) fn try_walker_specialize_subscr<Sym: WalkSym>(
...
+                        (*obj).w_class,
+                        pyre_object::get_instantiate(&pyre_object::INT_TYPE),
+                    )
+            }
+        {
+            None
+        } else {
+            Some(unsafe { pyre_object::w_int_get_value(obj) })
+        }
+    };
```
[the remark this answers is not in the document]
Comment (youknowone) [reply pyre/pyre-jit-trace/src/jitcode_dispatch/specialize.rs:8695]: Fixed in dba69287b7e. `exact_int_bound` now takes the `CAN_BE_TAGGED && is_tagged_int` precheck before any `ob_type` read, matching the sibling sites in this crate.

Scoped deliberately to the bounds: `is_slice(key_obj)` at the call site already filters a tagged key (`py_type_check` short-circuits immediates and returns false for a non-`INT_TYPE` target), and the `(*list_obj).ob_type` read that computes `tuple_canonical` predates this PR. The slice bounds were the one pair this change newly exposed — they arrive as raw stored fields with no type test upstream, so they were the only new hazard.

`CAN_BE_TAGGED` is false, so behaviour is unchanged; the guard is for whoever flips it.

— *commented by Claude*

File: pyre/pyre-jit-trace/src/jitcode_dispatch/specialize.rs [hunk 2]
```diff
@@ -8617,6 +8625,189 @@ pub(crate) fn try_walker_specialize_subscr<Sym: WalkSym>(
...
+/// result allocation live.
+///
+/// This first slice is intentionally the representation-complete arity-two
+/// arm: exact builtin slice, exact-int/None bounds, `step is None`, and exactly
+/// two selected elements. The authentic residual supplies the concrete result
+/// and verifies it chose `W_SpecialisedTupleObject_oo`; the emitted result uses
+/// that same layout. Every wider, empty, stepped, dynamic, subclass or
+/// non-exact-bound shape remains on `descr_getslice`.
+#[allow(clippy::too_many_arguments)]
+pub(crate) fn try_walker_specialize_subscr_tuple_slice2<Sym: WalkSym>(
```
[the remark this answers is not in the document]
Comment (youknowone) [reply pyre/pyre-jit-trace/src/jitcode_dispatch/specialize.rs:8646]: Partly taken, in dba69287b7e — but not as a translator/codewriter change.

Replacing the walker-native fold would equally condemn the ~90 other `try_walker_*` folds in this file, including the sibling `try_walker_specialize_subscr_tuple` directly below it. That shape is the established one here, not something this PR introduces.

The concrete risk you name is real, and narrower than the remedy: nothing tied the *emitted* element loads back to the interpreter's answer. `item0_concrete`/`item1_concrete` were read at `start` and `start + 1` from the local `clamp`, then attached as concretes to the IR reads — but never compared against `result_concrete`. If `clamp` ever drifted from `indices4`, the tracing iteration would still be correct while the compiled trace read the wrong indices, i.e. divergence visible only in compiled execution, which is exactly the failure class you describe.

That window is now closed: the fold compares both elements against the result `descr_getslice` actually produced and declines on disagreement, so the executed result is the authority on which elements may be named. `result_concrete` is already guarded to `SPECIALISED_TUPLE_OO_TYPE`, so this costs two loads and a pointer compare, and it agrees by construction whenever the indices are right — `exc_info_direct_hot`'s `spec-folds` header still reports both folds firing, and the full synthetic corpus is 522/522 on dynasm.

— *commented by Claude*
